playbooks/aws/openshift-cluster/provision.yml

```diff
@@ -0,0 +1,264 @@
+    - "'already exists' not in elb_cert_chain.msg"
+
+  - name: create our master external and internal load balancers
+    include_role:
+      name: openshift_aws_elb
+    vars:
+      r_openshift_aws_elb_clusterid: "{{ provision.clusterid }}"
+      r_openshift_aws_elb_region: "{{ provision.region }}"
+      r_openshift_aws_elb_instance_filter:
+        "tag:clusterid": "{{ provision.clusterid }}"
+        "tag:host-type": master
+        instance-state-name: running
+      r_openshift_aws_elb_type: master
+      r_openshift_aws_elb_direction: "{{ elb_item }}"
+      r_openshift_aws_elb_idle_timout: 400
+      r_openshift_aws_elb_scheme: internet-facing
+      r_openshift_aws_elb_security_groups:
+      - "{{ provision.clusterid }}"
+      - "{{ provision.clusterid }}_master"
+      r_openshift_aws_elb_subnet_name: "{{ provision.vpc.subnets[provision.region][0].az }}"
+      r_openshift_aws_elb_name: "{{ provision.clusterid }}-master-{{ elb_item }}"
+      r_openshift_aws_elb_cert_arn: "{{ provision.cert_arn }}"
+    with_items:
+    - internal
+    - external
+    loop_control:
+      loop_var: elb_item
+
+  - name: add new master to masters group
+    add_host:
+      groups: "{{ 'tag_clusterid_' ~ provision.clusterid }},masters,{{ 'tag_environment_' ~ provision.node_group_config.tags.environment }},tag_host-type_etcd,tag_host-type_master,tag_host-type_node"
+      name: "{{ item.public_ip_address }}"
+      hostname: "{{ provision.clusterid }}-master-{{ item.id[:-5] }}"
+    with_items: "{{ instancesout.instances }}"
+
+  - name: set facts for group normalization
+    set_fact:
+      cluster_id: "{{ provision.clusterid }}"
+      cluster_env: "{{ provision.node_group_config.tags.environment | default('dev') }}"
+
+  - name: normalize groups
+    include_vars: cluster_hosts.yml
+
+  - name: wait for ssh to become available
+    wait_for:
+      port: 22
+      host: "{{ item.public_ip_address }}"
+      timeout: 300
+      search_regex: OpenSSH
+    with_items: "{{ instancesout.instances }}"
+
+- name: set the master facts for hostname to elb
+  hosts: masters
+  gather_facts: no
+  remote_user: root
+  tasks:
+  - name: include vars
+    include_vars: vars.yml
+
+  - name: fetch elbs
+    ec2_elb_facts:
+      region: "{{ provision.region }}"
+      names:
+      - "{{ item }}"
+    with_items:
+    - "{{ provision.clusterid }}-master-external"
+    - "{{ provision.clusterid }}-master-internal"
+    delegate_to: localhost
+    register: elbs
+
+  - debug: var=elbs
+
+  - name: set fact
+    set_fact:
+      openshift_master_cluster_hostname: "{{ elbs.results[1].elbs[0].dns_name }}"
+      osm_custom_cors_origins:
+      - "{{ elbs.results[1].elbs[0].dns_name }}"
+      - "console.{{ provision.clusterid }}.openshift.com"
+      - "api.{{ provision.clusterid }}.openshift.com"
+    with_items: "{{ groups['masters'] }}"
+
+  - name:
+    debug: var=hostvars
+
+- include: ../../common/openshift-cluster/config.yml
```

Comment:
I'm still trying to figured out all the different paths through the plays, but this looks like a potential spot to break this play up into 3 (or more) plays:

```
#provision-site.yml
include <this play.yml>
include .....config.yml
include <the rest of this play.yml>
include <what happens next>
```

Just my opinion, fwiw.